### PR TITLE
Feature: Add and Delete passengers from a Booking

### PR DESCRIPTION
**Pull request by aalises**
Adding and deleting Passengers is now possible. Relates to #771  

*Validation: Name and ID are mandatory, and the ID has to be unique*. 

May-15-2019 15-03-03

### CODE REVIEW

File: packages/components/src/passengerCard/__tests__/__snapshots__/PassengerCard.test.js.snap [hunk 1]
```diff
@@ -146,7 +146,7 @@ Object {
         <PassengerCardDetail
           label="Date of birth"
           style="normal"
-          value="22/04/1980"
+          value="1980-04-12"
```
Comment (JoseDuSV) [inline packages/components/src/passengerCard/__tests__/__snapshots__/PassengerCard.test.js.snap]: Is this change related to format used in MMB? I think that so far we are using `DD/MM/YYYY` everywhere and it will be reasonable to keep it unified.

Comment (aalises) [reply packages/components/src/passengerCard/__tests__/__snapshots__/PassengerCard.test.js.snap]: @JoseDuSV Sure! I just didn't recall which format was in the config 😆So I put the first one

**Comment by FilipMessa** [comment]
And what about the edit? It will be in the next PR?

**Comment by aalises** [comment]
@FilipMessa Yes! That was the plan, to get this right as the edit will be done similarly.

**Comment by FilipMessa** [comment]
OK, so in that case, there should not be a comment for close the issue. 
I'll check the code

File: apps/core/scenes/resultDetail/resultDetailContent/ResultDetailPassenger.js [hunk 2]
```diff
@@ -1,53 +1,108 @@
...
 |};
 
 type State = {|
   +isFormVisible: boolean,
+  +passengerData: Array<PassengerCardType>,
 |};
 
-// @TODO - temp passengers info, load default values from user profile or set empty for uregistered
-const passengerCards = [
+const defaultPassengers = [
```
Comment (FilipMessa) [inline apps/core/scenes/resultDetail/resultDetailContent/ResultDetailPassenger.js]: this should not be mocked in future som pls add some TODO comment there, THANKS

File: apps/core/scenes/resultDetail/resultDetailContent/ResultDetailPassenger.js [hunk 3]
```diff
@@ -1,53 +1,108 @@
...
-    this.setState({ isFormVisible: false });
+  handleFormSaveRequest = passenger => {
+    const existsID: boolean = !!this.state.passengerData.find(
+      el => el.id === passenger.id,
+    );
+    const validateForm = () => {
+      if (!(passenger.id && passenger.name)) {
+        return 'ID and Name are mandatory';
+      } else if (existsID) {
+        return 'A passenger with the same ID has already been added. Please choose another one';
```
Comment (FilipMessa) [inline apps/core/scenes/resultDetail/resultDetailContent/ResultDetailPassenger.js]:
```suggestion
        return 'A passenger with the same ID has already been added.';
```

File: packages/components/src/passengerCard/PassengerCard.js [hunk 4]
```diff
@@ -64,13 +72,15 @@ class PassengerCard extends React.Component<Props> {
       passengerCount,
       bags,
       actionIconName,
+      secondaryActionIconName,
       onActionPress,
+      onSecondaryActionPress,
```
Comment (FilipMessa) [inline packages/components/src/passengerCard/PassengerCard.js]: Are we using this card for something else? Can we just rename it `onEditPress` and `onDeletePress`?

Comment (aalises) [reply packages/components/src/passengerCard/PassengerCard.js]: All done!

Comment (FilipMessa) [reply packages/components/src/passengerCard/PassengerCard.js]: Thanks!
